### PR TITLE
feat: add prometheus rule support

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds prometheus rule support.

It is complementary with `serviceMonitor`.

### CODE REVIEW

**Comment by gjtempleton** [comment]
Hi, thanks for the PR!

Can you please make a minor version bump to the chart's `version` under `Chart.yml` as well.

**Comment by eddycharly** [comment]
Sure, I bumped the version to `9.6.0`.

File: charts/cluster-autoscaler/values.yaml [hunk 1]
```diff
@@ -298,6 +298,19 @@ serviceMonitor:
...
+## The value is evaluated as a template, so, for example, the value can depend on .Release or .Chart
+## ref: https://github.com/coreos/prometheus-operator#customresourcedefinitions
+prometheusRule:
+  # prometheusRule.enabled -- If true, creates a Prometheus Operator PrometheusRule.
+  enabled: false
+  # prometheusRule.additionalLabels -- Additional labels to be set in metadata.
+  additionalLabels: {}
+  # prometheusRule.namespace -- Namespace which Prometheus is running in.
+  namespace: monitoring
+  # prometheusRule.rules -- Rules spec template.
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/values.yaml:311]: Is it worth linking to the API spec for the rule object here? I.e. https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#rule

File: charts/cluster-autoscaler/templates/prometheusrule.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ include "cluster-autoscaler.fullname" . }}
+  {{- if .Values.prometheusRule.namespace }}
+  namespace: {{ .Values.prometheusRule.namespace }}
+  {{- end }}
+  labels: {{- toYaml .Values.prometheusRule.additionalLabels | nindent 4 }}
+spec:
+  groups:
+    - name: {{ include "cluster-autoscaler.fullname" . }}
+      rules: {{- tpl (toYaml .Values.prometheusRule.rules) . | nindent 8 }}
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/templates/prometheusrule.yaml:13]: Do we want to support users setting the interval as well? (Potentially the partial_response_strategy as well, though that's probably less useful, so not as sold on including.) 

https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#rulegroup

Comment (eddycharly) [reply charts/cluster-autoscaler/templates/prometheusrule.yaml:13]: Good catch ! , i'll add that.

File: charts/cluster-autoscaler/values.yaml (hunk 1 above)
Comment (eddycharly) [reply charts/cluster-autoscaler/values.yaml]: Added the reference doc link. I'm thinking about adding an example too, what do you think ?

File: charts/cluster-autoscaler/templates/prometheusrule.yaml (hunk 2 above)
Comment (eddycharly) [reply charts/cluster-autoscaler/templates/prometheusrule.yaml]: Done for `interval`. Not sure about `partial_response_strategy` either.

**Pull request merged**